### PR TITLE
Fix internal error

### PR DESCRIPTION
**Pull request by turtlewizard73**
[TRELLO CARD](https://trello.com/c/VsIJoTDk/628-investigate-and-fix-s2-lidar-driver-crash)

### CODE REVIEW

File: launch/view_sllidar_t1_launch.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def generate_launch_description():
     frame_id = LaunchConfiguration('frame_id', default='laser')
     inverted = LaunchConfiguration('inverted', default='false')
     angle_compensate = LaunchConfiguration('angle_compensate', default='true')
-    scan_mode = LaunchConfiguration('scan_mode', default='DenseBoost')
+    scan_mode = LaunchConfiguration('scan_mode', default='Sensitivity')
```
Comment (emilnovak) [inline launch/view_sllidar_t1_launch.py:20]: This may not needed in this pr

File: src/sllidar_node.cpp [hunk 2]
```diff
@@ -138,14 +138,24 @@ class SLlidarNode : public rclcpp::Node
...
             return false;
         }
     }
 
+    bool manage_internal_error()
+    {
+        RCLCPP_INFO(this->get_logger(),"Reseting SLLidar");
+        sl_result reset_result;
+        reset_result = drv->reset();
+        RCLCPP_INFO(this->get_logger(),"Reset ok?: %x", reset_result);
```
Comment (emilnovak) [inline src/sllidar_node.cpp]: If there is a chance that the reset fails, than add n tries and return false if that happens

File: src/sllidar_node.cpp [hunk 3]
```diff
@@ -138,14 +138,24 @@ class SLlidarNode : public rclcpp::Node
                     return true;
                 case SL_LIDAR_STATUS_ERROR:
                     RCLCPP_ERROR(this->get_logger(),"Error, SLLidar internal error detected. Please reboot the device to retry.");
-                    return false;
+                    return manage_internal_error();
+                    // return false;
```
Comment (emilnovak) [inline src/sllidar_node.cpp]: This line is no longer needed

**Comment by RBT22** [comment]
initialize task succeeded even with node errors

**Pull request merged**